packages/next-codemod/bin/upgrade.ts

```diff
@@ -30,6 +30,71 @@ async function loadHighestNPMVersionMatching(query: string) {
   return versions[versions.length - 1]
 }
 
+const codeSnippet = (code: string) => chalk.grey(`${code}`)
+
+const MIGRATION_INSTRUCTIONS = `\
+The codemods are applied, please review the changes of transformed files and follow the instructions to complete the migration:
+
+- Search for comment "${chalk.white('Next.js Dynamic Async API Codemod:')}", and read the instructions in comment and migrate them manually.
+  For example:
+
+  Codemod output:${codeSnippet(`
+  \`\`\`
+  import { cookies } from 'next/headers'
+  export function foo() {
+    callback(/* Next.js Dynamic Async API Codemod: 'props' is passed as an argument. 
+    Any asynchronous properties of 'props' must be awaited when accessed. */
+    cookies())
+  }
+  \`\`\``)}
+
+  Manual migration:${codeSnippet(`
+  \`\`\`
+  import { cookies } from 'next/headers'
+  export async function foo() {
+    callback(await cookies())
+  }
+  \`\`\``)}
+  And update the usage of \`foo()\` to make sure it's awaited if necessary.    
+
+  
+- Search for Dynamic API type cast, replace the type casting with awaiting the API calls, and populate the async scope.
```

Comment:
- "Search for the newly added type casts, UnsafeUnwrappedCookies, etc" and refactor the code accordingly to make it async